common/arcus-model/platform-client/src/main/java/com/iris/client/impl/netty/HttpRequester.java

```diff
@@ -241,16 +241,18 @@ protected void sendErrorToPending(Throwable cause) {
    
    private class HttpRequestInitializer extends ChannelInitializer<SocketChannel> {
       private final SslContext sslCtx;
-      
-      HttpRequestInitializer(SslContext sslCtx) {
+      private final URI uri;
+
+       HttpRequestInitializer(SslContext sslCtx, URI u) {
          this.sslCtx = sslCtx;
+         this.uri = u;
       }
 
       @Override
       protected void initChannel(SocketChannel ch) throws Exception {
          ChannelPipeline pipeline = ch.pipeline();
          if (sslCtx != null) {
```

Comment:
should uri also have a null check?

Reply:
Code will not reach there with a null URI. Better would be to add the @NotNull annotation to the URI argument in the constructor for HttpRequester.